2022-12-13 00:54:38 database: flush
org.h2.message.DbException: Внутренняя ошибка: "org.h2.mvstore.MVStoreException: The file is locked: D:/programs/Java/learningprogresstracking/TestDatabase/test.mv.db [2.1.214/7]"
General error: "org.h2.mvstore.MVStoreException: The file is locked: D:/programs/Java/learningprogresstracking/TestDatabase/test.mv.db [2.1.214/7]" [50000-214]
	at org.h2.message.DbException.get(DbException.java:212)
	at org.h2.message.DbException.convert(DbException.java:395)
	at org.h2.mvstore.db.Store.lambda$new$0(Store.java:125)
	at org.h2.mvstore.MVStore.handleException(MVStore.java:3318)
	at org.h2.mvstore.MVStore.panic(MVStore.java:593)
	at org.h2.mvstore.MVStore.<init>(MVStore.java:469)
	at org.h2.mvstore.MVStore$Builder.open(MVStore.java:4082)
	at org.h2.mvstore.db.Store.<init>(Store.java:136)
	at org.h2.engine.ru.croc.project.database.<init>(ru.croc.project.database.java:324)
	at org.h2.engine.Engine.openSession(Engine.java:92)
	at org.h2.engine.Engine.openSession(Engine.java:222)
	at org.h2.engine.Engine.createSession(Engine.java:201)
	at org.h2.engine.SessionRemote.connectEmbeddedOrServer(SessionRemote.java:338)
	at org.h2.jdbc.JdbcConnection.<init>(JdbcConnection.java:122)
	at org.h2.Driver.connect(Driver.java:59)
	at java.sql/java.sql.DriverManager.getConnection(DriverManager.java:683)
	at java.sql/java.sql.DriverManager.getConnection(DriverManager.java:230)
	at ru.croc.project.database.Main.main(ru.croc.project.database.Main.java:11)
Caused by: org.h2.jdbc.JdbcSQLNonTransientException: Внутренняя ошибка: "org.h2.mvstore.MVStoreException: The file is locked: D:/programs/Java/learningprogresstracking/TestDatabase/test.mv.db [2.1.214/7]"
General error: "org.h2.mvstore.MVStoreException: The file is locked: D:/programs/Java/learningprogresstracking/TestDatabase/test.mv.db [2.1.214/7]" [50000-214]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:554)
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:477)
	... 18 more
Caused by: org.h2.mvstore.MVStoreException: The file is locked: D:/programs/Java/learningprogresstracking/TestDatabase/test.mv.db [2.1.214/7]
	at org.h2.mvstore.DataUtils.newMVStoreException(DataUtils.java:1004)
	at org.h2.mvstore.FileStore.open(FileStore.java:178)
	at org.h2.mvstore.FileStore.open(FileStore.java:128)
	at org.h2.mvstore.MVStore.<init>(MVStore.java:452)
	... 12 more
2022-12-13 01:02:23 database: flush
org.h2.message.DbException: Внутренняя ошибка: "org.h2.mvstore.MVStoreException: The file is locked: D:/programs/Java/learningprogresstracking/TestDatabase/test.mv.db [2.1.214/7]"
General error: "org.h2.mvstore.MVStoreException: The file is locked: D:/programs/Java/learningprogresstracking/TestDatabase/test.mv.db [2.1.214/7]" [50000-214]
	at org.h2.message.DbException.get(DbException.java:212)
	at org.h2.message.DbException.convert(DbException.java:395)
	at org.h2.mvstore.db.Store.lambda$new$0(Store.java:125)
	at org.h2.mvstore.MVStore.handleException(MVStore.java:3318)
	at org.h2.mvstore.MVStore.panic(MVStore.java:593)
	at org.h2.mvstore.MVStore.<init>(MVStore.java:469)
	at org.h2.mvstore.MVStore$Builder.open(MVStore.java:4082)
	at org.h2.mvstore.db.Store.<init>(Store.java:136)
	at org.h2.engine.ru.croc.project.database.<init>(ru.croc.project.database.java:324)
	at org.h2.engine.Engine.openSession(Engine.java:92)
	at org.h2.engine.Engine.openSession(Engine.java:222)
	at org.h2.engine.Engine.createSession(Engine.java:201)
	at org.h2.engine.SessionRemote.connectEmbeddedOrServer(SessionRemote.java:338)
	at org.h2.jdbc.JdbcConnection.<init>(JdbcConnection.java:122)
	at org.h2.Driver.connect(Driver.java:59)
	at java.sql/java.sql.DriverManager.getConnection(DriverManager.java:683)
	at java.sql/java.sql.DriverManager.getConnection(DriverManager.java:230)
	at ru.croc.project.database.Main.main(ru.croc.project.database.Main.java:11)
Caused by: org.h2.jdbc.JdbcSQLNonTransientException: Внутренняя ошибка: "org.h2.mvstore.MVStoreException: The file is locked: D:/programs/Java/learningprogresstracking/TestDatabase/test.mv.db [2.1.214/7]"
General error: "org.h2.mvstore.MVStoreException: The file is locked: D:/programs/Java/learningprogresstracking/TestDatabase/test.mv.db [2.1.214/7]" [50000-214]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:554)
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:477)
	... 18 more
Caused by: org.h2.mvstore.MVStoreException: The file is locked: D:/programs/Java/learningprogresstracking/TestDatabase/test.mv.db [2.1.214/7]
	at org.h2.mvstore.DataUtils.newMVStoreException(DataUtils.java:1004)
	at org.h2.mvstore.FileStore.open(FileStore.java:178)
	at org.h2.mvstore.FileStore.open(FileStore.java:128)
	at org.h2.mvstore.MVStore.<init>(MVStore.java:452)
	... 12 more
2022-12-13 01:06:09 database: flush
org.h2.message.DbException: Внутренняя ошибка: "org.h2.mvstore.MVStoreException: The file is locked: D:/programs/Java/learningprogresstracking/TestDatabase/test.mv.db [2.1.214/7]"
General error: "org.h2.mvstore.MVStoreException: The file is locked: D:/programs/Java/learningprogresstracking/TestDatabase/test.mv.db [2.1.214/7]" [50000-214]
	at org.h2.message.DbException.get(DbException.java:212)
	at org.h2.message.DbException.convert(DbException.java:395)
	at org.h2.mvstore.db.Store.lambda$new$0(Store.java:125)
	at org.h2.mvstore.MVStore.handleException(MVStore.java:3318)
	at org.h2.mvstore.MVStore.panic(MVStore.java:593)
	at org.h2.mvstore.MVStore.<init>(MVStore.java:469)
	at org.h2.mvstore.MVStore$Builder.open(MVStore.java:4082)
	at org.h2.mvstore.db.Store.<init>(Store.java:136)
	at org.h2.engine.ru.croc.project.database.<init>(ru.croc.project.database.java:324)
	at org.h2.engine.Engine.openSession(Engine.java:92)
	at org.h2.engine.Engine.openSession(Engine.java:222)
	at org.h2.engine.Engine.createSession(Engine.java:201)
	at org.h2.engine.SessionRemote.connectEmbeddedOrServer(SessionRemote.java:338)
	at org.h2.jdbc.JdbcConnection.<init>(JdbcConnection.java:122)
	at org.h2.Driver.connect(Driver.java:59)
	at java.sql/java.sql.DriverManager.getConnection(DriverManager.java:683)
	at java.sql/java.sql.DriverManager.getConnection(DriverManager.java:230)
	at ru.croc.project.database.Main.main(ru.croc.project.database.Main.java:11)
Caused by: org.h2.jdbc.JdbcSQLNonTransientException: Внутренняя ошибка: "org.h2.mvstore.MVStoreException: The file is locked: D:/programs/Java/learningprogresstracking/TestDatabase/test.mv.db [2.1.214/7]"
General error: "org.h2.mvstore.MVStoreException: The file is locked: D:/programs/Java/learningprogresstracking/TestDatabase/test.mv.db [2.1.214/7]" [50000-214]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:554)
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:477)
	... 18 more
Caused by: org.h2.mvstore.MVStoreException: The file is locked: D:/programs/Java/learningprogresstracking/TestDatabase/test.mv.db [2.1.214/7]
	at org.h2.mvstore.DataUtils.newMVStoreException(DataUtils.java:1004)
	at org.h2.mvstore.FileStore.open(FileStore.java:178)
	at org.h2.mvstore.FileStore.open(FileStore.java:128)
	at org.h2.mvstore.MVStore.<init>(MVStore.java:452)
	... 12 more
2022-12-13 01:06:46 jdbc[3]: exception
org.h2.jdbc.JdbcSQLSyntaxErrorException: Синтаксическая ошибка в выражении SQL "CREATE TABLE Product(\000d\000a                              code VARCHAR(255) PRIMARY KEY,\000d\000a                              name VARCHAR(255),\000d\000a                              price INT[*]"; ожидалось "ARRAY, INVISIBLE, VISIBLE, NOT NULL, NULL, AS, DEFAULT, GENERATED, ON UPDATE, NOT NULL, NULL, AUTO_INCREMENT, DEFAULT ON NULL, NULL_TO_DEFAULT, SEQUENCE, SELECTIVITY, COMMENT, CONSTRAINT, COMMENT, PRIMARY KEY, UNIQUE, NOT NULL, NULL, CHECK, REFERENCES, AUTO_INCREMENT, ,, )"
Syntax error in SQL statement "CREATE TABLE Product(\000d\000a                              code VARCHAR(255) PRIMARY KEY,\000d\000a                              name VARCHAR(255),\000d\000a                              price INT[*]"; expected "ARRAY, INVISIBLE, VISIBLE, NOT NULL, NULL, AS, DEFAULT, GENERATED, ON UPDATE, NOT NULL, NULL, AUTO_INCREMENT, DEFAULT ON NULL, NULL_TO_DEFAULT, SEQUENCE, SELECTIVITY, COMMENT, CONSTRAINT, COMMENT, PRIMARY KEY, UNIQUE, NOT NULL, NULL, CHECK, REFERENCES, AUTO_INCREMENT, ,, )"; SQL statement:
CREATE TABLE Product(
                              code VARCHAR(255) PRIMARY KEY,
                              name VARCHAR(255),
                              price INT [42001-214]
2022-12-13 02:35:18 jdbc[3]: exception
org.h2.jdbc.JdbcSQLSyntaxErrorException: Синтаксическая ошибка в выражении SQL "CREATE TABLE UserStats(\000d\000a\0009id INT PRIMARY KEY AUTO_INCREMENT,\000d\000a\0009username VARCHAR(255) NOT NULL,\000d\000a\0009wordsLearned INT,\000d\000a\0009testsCompleted INT,\000d\000a\0009favouriteWord VARCHAR(255),\000d\000a\0009problemWord VARCHAR(255),\000d\000a[*])"; ожидалось "identifier"
Syntax error in SQL statement "CREATE TABLE UserStats(\000d\000a\0009id INT PRIMARY KEY AUTO_INCREMENT,\000d\000a\0009username VARCHAR(255) NOT NULL,\000d\000a\0009wordsLearned INT,\000d\000a\0009testsCompleted INT,\000d\000a\0009favouriteWord VARCHAR(255),\000d\000a\0009problemWord VARCHAR(255),\000d\000a[*])"; expected "identifier"; SQL statement:
CREATE TABLE UserStats(
	id INT PRIMARY KEY AUTO_INCREMENT,
	username VARCHAR(255) NOT NULL,
	wordsLearned INT,
	testsCompleted INT,
	favouriteWord VARCHAR(255),
	problemWord VARCHAR(255),
) [42001-214]
2022-12-13 02:48:55 jdbc[3]: exception
org.h2.jdbc.JdbcSQLSyntaxErrorException: Повтор имени столбца "WORDSCOUNT"
Duplicate column name "WORDSCOUNT"; SQL statement:
ALTER TABLE USERSTATS ADD WORDSCOUNT VARCHAR(255) [42121-214]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:502)
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:477)
	at org.h2.message.DbException.get(DbException.java:223)
	at org.h2.message.DbException.get(DbException.java:199)
	at org.h2.table.Table.setColumns(Table.java:487)
	at org.h2.table.TableBase.<init>(TableBase.java:73)
	at org.h2.mvstore.db.MVTable.<init>(MVTable.java:139)
	at org.h2.mvstore.db.Store.createTable(Store.java:219)
	at org.h2.schema.Schema.createTable(Schema.java:776)
	at org.h2.command.ddl.AlterTableAlterColumn.cloneTableStructure(AlterTableAlterColumn.java:454)
	at org.h2.command.ddl.AlterTableAlterColumn.copyData(AlterTableAlterColumn.java:339)
	at org.h2.command.ddl.AlterTableAlterColumn.update(AlterTableAlterColumn.java:242)
	at org.h2.command.CommandContainer.update(CommandContainer.java:169)
	at org.h2.command.Command.executeUpdate(Command.java:252)
	at org.h2.jdbc.JdbcStatement.executeInternal(JdbcStatement.java:252)
	at org.h2.jdbc.JdbcStatement.execute(JdbcStatement.java:223)
	at org.h2.server.web.WebApp.getResult(WebApp.java:1339)
	at org.h2.server.web.WebApp.query(WebApp.java:1137)
	at org.h2.server.web.WebApp$1.next(WebApp.java:1103)
	at org.h2.server.web.WebApp$1.next(WebApp.java:1090)
	at org.h2.server.web.WebThread.process(WebThread.java:188)
	at org.h2.server.web.WebThread.run(WebThread.java:101)
	at java.base/java.lang.Thread.run(Thread.java:833)
2022-12-13 03:03:03 jdbc[3]: exception
org.h2.jdbc.JdbcSQLSyntaxErrorException: Синтаксическая ошибка в выражении SQL "ALTER TABLE [*]? ADD ? ?"; ожидалось "identifier"
Syntax error in SQL statement "ALTER TABLE [*]? ADD ? ?"; expected "identifier"; SQL statement:
ALTER TABLE ? ADD ? ? [42001-214]
2022-12-13 03:04:32 jdbc[3]: exception
org.h2.jdbc.JdbcSQLSyntaxErrorException: Синтаксическая ошибка в выражении SQL "ALTER TABLE rs0[*]: org.h2.result.SimpleResult@55040f2f ADD ? ?"; ожидалось "., ADD, SET, RENAME, DROP, ALTER"
Syntax error in SQL statement "ALTER TABLE rs0[*]: org.h2.result.SimpleResult@55040f2f ADD ? ?"; expected "., ADD, SET, RENAME, DROP, ALTER"; SQL statement:
ALTER TABLE rs0: org.h2.result.SimpleResult@55040f2f ADD ? ? [42001-214]
2022-12-13 03:05:34 jdbc[3]: exception
org.h2.jdbc.JdbcSQLSyntaxErrorException: Синтаксическая ошибка в выражении SQL "ALTER TABLE rs0[*]: org.h2.result.SimpleResult@55040f2f ADD ? ?"; ожидалось "., ADD, SET, RENAME, DROP, ALTER"
Syntax error in SQL statement "ALTER TABLE rs0[*]: org.h2.result.SimpleResult@55040f2f ADD ? ?"; expected "., ADD, SET, RENAME, DROP, ALTER"; SQL statement:
ALTER TABLE rs0: org.h2.result.SimpleResult@55040f2f ADD ? ? [42001-214]
2022-12-13 03:06:14 jdbc[3]: exception
org.h2.jdbc.JdbcSQLSyntaxErrorException: Синтаксическая ошибка в выражении SQL "ALTER TABLE USERSTATS ADD [*]? ?"; ожидалось "identifier"
Syntax error in SQL statement "ALTER TABLE USERSTATS ADD [*]? ?"; expected "identifier"; SQL statement:
ALTER TABLE USERSTATS ADD ? ? [42001-214]
2022-12-13 03:08:05 database: flush
org.h2.message.DbException: Внутренняя ошибка: "org.h2.mvstore.MVStoreException: The file is locked: D:/programs/Java/learningprogresstracking/TestDatabase/test.mv.db [2.1.214/7]"
General error: "org.h2.mvstore.MVStoreException: The file is locked: D:/programs/Java/learningprogresstracking/TestDatabase/test.mv.db [2.1.214/7]" [50000-214]
	at org.h2.message.DbException.get(DbException.java:212)
	at org.h2.message.DbException.convert(DbException.java:395)
	at org.h2.mvstore.db.Store.lambda$new$0(Store.java:125)
	at org.h2.mvstore.MVStore.handleException(MVStore.java:3318)
	at org.h2.mvstore.MVStore.panic(MVStore.java:593)
	at org.h2.mvstore.MVStore.<init>(MVStore.java:469)
	at org.h2.mvstore.MVStore$Builder.open(MVStore.java:4082)
	at org.h2.mvstore.db.Store.<init>(Store.java:136)
	at org.h2.engine.Database.<init>(Database.java:324)
	at org.h2.engine.Engine.openSession(Engine.java:92)
	at org.h2.engine.Engine.openSession(Engine.java:222)
	at org.h2.engine.Engine.createSession(Engine.java:201)
	at org.h2.engine.SessionRemote.connectEmbeddedOrServer(SessionRemote.java:338)
	at org.h2.jdbc.JdbcConnection.<init>(JdbcConnection.java:122)
	at org.h2.Driver.connect(Driver.java:59)
	at java.sql/java.sql.DriverManager.getConnection(DriverManager.java:683)
	at java.sql/java.sql.DriverManager.getConnection(DriverManager.java:230)
	at ru.croc.project.database.access.H2Database.createConnection(H2Database.java:18)
	at ru.croc.project.database.UserDatabaseController.initDatabase(UserDatabaseController.java:35)
	at ru.croc.project.database.Main.main(Main.java:8)
Caused by: org.h2.jdbc.JdbcSQLNonTransientException: Внутренняя ошибка: "org.h2.mvstore.MVStoreException: The file is locked: D:/programs/Java/learningprogresstracking/TestDatabase/test.mv.db [2.1.214/7]"
General error: "org.h2.mvstore.MVStoreException: The file is locked: D:/programs/Java/learningprogresstracking/TestDatabase/test.mv.db [2.1.214/7]" [50000-214]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:554)
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:477)
	... 20 more
Caused by: org.h2.mvstore.MVStoreException: The file is locked: D:/programs/Java/learningprogresstracking/TestDatabase/test.mv.db [2.1.214/7]
	at org.h2.mvstore.DataUtils.newMVStoreException(DataUtils.java:1004)
	at org.h2.mvstore.FileStore.open(FileStore.java:178)
	at org.h2.mvstore.FileStore.open(FileStore.java:128)
	at org.h2.mvstore.MVStore.<init>(MVStore.java:452)
	... 14 more
2022-12-13 03:08:12 jdbc[3]: exception
org.h2.jdbc.JdbcSQLSyntaxErrorException: Синтаксическая ошибка в выражении SQL "ALTER TABLE [*]? ADD ? ?"; ожидалось "identifier"
Syntax error in SQL statement "ALTER TABLE [*]? ADD ? ?"; expected "identifier"; SQL statement:
ALTER TABLE ? ADD ? ? [42001-214]
2022-12-13 03:09:23 jdbc[3]: exception
org.h2.jdbc.JdbcSQLSyntaxErrorException: Синтаксическая ошибка в выражении SQL "ALTER TABLE USERSTATS ADD [*]? ?"; ожидалось "identifier"
Syntax error in SQL statement "ALTER TABLE USERSTATS ADD [*]? ?"; expected "identifier"; SQL statement:
ALTER TABLE USERSTATS ADD ? ? [42001-214]
2022-12-13 03:11:45 database: flush
org.h2.message.DbException: Внутренняя ошибка: "org.h2.mvstore.MVStoreException: The file is locked: D:/programs/Java/learningprogresstracking/TestDatabase/test.mv.db [2.1.214/7]"
General error: "org.h2.mvstore.MVStoreException: The file is locked: D:/programs/Java/learningprogresstracking/TestDatabase/test.mv.db [2.1.214/7]" [50000-214]
	at org.h2.message.DbException.get(DbException.java:212)
	at org.h2.message.DbException.convert(DbException.java:395)
	at org.h2.mvstore.db.Store.lambda$new$0(Store.java:125)
	at org.h2.mvstore.MVStore.handleException(MVStore.java:3318)
	at org.h2.mvstore.MVStore.panic(MVStore.java:593)
	at org.h2.mvstore.MVStore.<init>(MVStore.java:469)
	at org.h2.mvstore.MVStore$Builder.open(MVStore.java:4082)
	at org.h2.mvstore.db.Store.<init>(Store.java:136)
	at org.h2.engine.Database.<init>(Database.java:324)
	at org.h2.engine.Engine.openSession(Engine.java:92)
	at org.h2.engine.Engine.openSession(Engine.java:222)
	at org.h2.engine.Engine.createSession(Engine.java:201)
	at org.h2.engine.SessionRemote.connectEmbeddedOrServer(SessionRemote.java:338)
	at org.h2.jdbc.JdbcConnection.<init>(JdbcConnection.java:122)
	at org.h2.Driver.connect(Driver.java:59)
	at java.sql/java.sql.DriverManager.getConnection(DriverManager.java:683)
	at java.sql/java.sql.DriverManager.getConnection(DriverManager.java:230)
	at ru.croc.project.database.access.H2Database.createConnection(H2Database.java:18)
	at ru.croc.project.database.UserDatabaseController.initDatabase(UserDatabaseController.java:35)
	at ru.croc.project.database.Main.main(Main.java:8)
Caused by: org.h2.jdbc.JdbcSQLNonTransientException: Внутренняя ошибка: "org.h2.mvstore.MVStoreException: The file is locked: D:/programs/Java/learningprogresstracking/TestDatabase/test.mv.db [2.1.214/7]"
General error: "org.h2.mvstore.MVStoreException: The file is locked: D:/programs/Java/learningprogresstracking/TestDatabase/test.mv.db [2.1.214/7]" [50000-214]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:554)
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:477)
	... 20 more
Caused by: org.h2.mvstore.MVStoreException: The file is locked: D:/programs/Java/learningprogresstracking/TestDatabase/test.mv.db [2.1.214/7]
	at org.h2.mvstore.DataUtils.newMVStoreException(DataUtils.java:1004)
	at org.h2.mvstore.FileStore.open(FileStore.java:178)
	at org.h2.mvstore.FileStore.open(FileStore.java:128)
	at org.h2.mvstore.MVStore.<init>(MVStore.java:452)
	... 14 more
2022-12-13 03:13:56 database: flush
org.h2.message.DbException: Внутренняя ошибка: "org.h2.mvstore.MVStoreException: The file is locked: D:/programs/Java/learningprogresstracking/TestDatabase/test.mv.db [2.1.214/7]"
General error: "org.h2.mvstore.MVStoreException: The file is locked: D:/programs/Java/learningprogresstracking/TestDatabase/test.mv.db [2.1.214/7]" [50000-214]
	at org.h2.message.DbException.get(DbException.java:212)
	at org.h2.message.DbException.convert(DbException.java:395)
	at org.h2.mvstore.db.Store.lambda$new$0(Store.java:125)
	at org.h2.mvstore.MVStore.handleException(MVStore.java:3318)
	at org.h2.mvstore.MVStore.panic(MVStore.java:593)
	at org.h2.mvstore.MVStore.<init>(MVStore.java:469)
	at org.h2.mvstore.MVStore$Builder.open(MVStore.java:4082)
	at org.h2.mvstore.db.Store.<init>(Store.java:136)
	at org.h2.engine.Database.<init>(Database.java:324)
	at org.h2.engine.Engine.openSession(Engine.java:92)
	at org.h2.engine.Engine.openSession(Engine.java:222)
	at org.h2.engine.Engine.createSession(Engine.java:201)
	at org.h2.engine.SessionRemote.connectEmbeddedOrServer(SessionRemote.java:338)
	at org.h2.jdbc.JdbcConnection.<init>(JdbcConnection.java:122)
	at org.h2.Driver.connect(Driver.java:59)
	at java.sql/java.sql.DriverManager.getConnection(DriverManager.java:683)
	at java.sql/java.sql.DriverManager.getConnection(DriverManager.java:230)
	at ru.croc.project.database.access.H2Database.createConnection(H2Database.java:18)
	at ru.croc.project.database.UserDatabaseController.clearDatabase(UserDatabaseController.java:64)
	at ru.croc.project.database.Test.main(Test.java:8)
Caused by: org.h2.jdbc.JdbcSQLNonTransientException: Внутренняя ошибка: "org.h2.mvstore.MVStoreException: The file is locked: D:/programs/Java/learningprogresstracking/TestDatabase/test.mv.db [2.1.214/7]"
General error: "org.h2.mvstore.MVStoreException: The file is locked: D:/programs/Java/learningprogresstracking/TestDatabase/test.mv.db [2.1.214/7]" [50000-214]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:554)
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:477)
	... 20 more
Caused by: org.h2.mvstore.MVStoreException: The file is locked: D:/programs/Java/learningprogresstracking/TestDatabase/test.mv.db [2.1.214/7]
	at org.h2.mvstore.DataUtils.newMVStoreException(DataUtils.java:1004)
	at org.h2.mvstore.FileStore.open(FileStore.java:178)
	at org.h2.mvstore.FileStore.open(FileStore.java:128)
	at org.h2.mvstore.MVStore.<init>(MVStore.java:452)
	... 14 more
2022-12-13 03:14:03 jdbc[3]: exception
org.h2.jdbc.JdbcSQLSyntaxErrorException: Синтаксическая ошибка в выражении SQL "DROP TABLE [*]?"; ожидалось "identifier"
Syntax error in SQL statement "DROP TABLE [*]?"; expected "identifier"; SQL statement:
DROP TABLE ? [42001-214]
2022-12-13 15:30:00 database: flush
org.h2.message.DbException: Внутренняя ошибка: "org.h2.mvstore.MVStoreException: The file is locked: D:/programs/Java/learningprogresstracking/TestDatabase/test.mv.db [2.1.214/7]"
General error: "org.h2.mvstore.MVStoreException: The file is locked: D:/programs/Java/learningprogresstracking/TestDatabase/test.mv.db [2.1.214/7]" [50000-214]
	at org.h2.message.DbException.get(DbException.java:212)
	at org.h2.message.DbException.convert(DbException.java:395)
	at org.h2.mvstore.db.Store.lambda$new$0(Store.java:125)
	at org.h2.mvstore.MVStore.handleException(MVStore.java:3318)
	at org.h2.mvstore.MVStore.panic(MVStore.java:593)
	at org.h2.mvstore.MVStore.<init>(MVStore.java:469)
	at org.h2.mvstore.MVStore$Builder.open(MVStore.java:4082)
	at org.h2.mvstore.db.Store.<init>(Store.java:136)
	at org.h2.engine.Database.<init>(Database.java:324)
	at org.h2.engine.Engine.openSession(Engine.java:92)
	at org.h2.engine.Engine.openSession(Engine.java:222)
	at org.h2.engine.Engine.createSession(Engine.java:201)
	at org.h2.engine.SessionRemote.connectEmbeddedOrServer(SessionRemote.java:338)
	at org.h2.jdbc.JdbcConnection.<init>(JdbcConnection.java:122)
	at org.h2.Driver.connect(Driver.java:59)
	at java.sql/java.sql.DriverManager.getConnection(DriverManager.java:683)
	at java.sql/java.sql.DriverManager.getConnection(DriverManager.java:230)
	at ru.croc.project.database.access.H2Database.createConnection(H2Database.java:18)
	at ru.croc.project.database.UserDatabaseController.clearDatabase(UserDatabaseController.java:43)
	at ru.croc.project.database.Test.main(Test.java:19)
Caused by: org.h2.jdbc.JdbcSQLNonTransientException: Внутренняя ошибка: "org.h2.mvstore.MVStoreException: The file is locked: D:/programs/Java/learningprogresstracking/TestDatabase/test.mv.db [2.1.214/7]"
General error: "org.h2.mvstore.MVStoreException: The file is locked: D:/programs/Java/learningprogresstracking/TestDatabase/test.mv.db [2.1.214/7]" [50000-214]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:554)
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:477)
	... 20 more
Caused by: org.h2.mvstore.MVStoreException: The file is locked: D:/programs/Java/learningprogresstracking/TestDatabase/test.mv.db [2.1.214/7]
	at org.h2.mvstore.DataUtils.newMVStoreException(DataUtils.java:1004)
	at org.h2.mvstore.FileStore.open(FileStore.java:178)
	at org.h2.mvstore.FileStore.open(FileStore.java:128)
	at org.h2.mvstore.MVStore.<init>(MVStore.java:452)
	... 14 more
2022-12-13 15:34:15 database: flush
org.h2.message.DbException: Внутренняя ошибка: "org.h2.mvstore.MVStoreException: The file is locked: D:/programs/Java/learningprogresstracking/TestDatabase/test.mv.db [2.1.214/7]"
General error: "org.h2.mvstore.MVStoreException: The file is locked: D:/programs/Java/learningprogresstracking/TestDatabase/test.mv.db [2.1.214/7]" [50000-214]
	at org.h2.message.DbException.get(DbException.java:212)
	at org.h2.message.DbException.convert(DbException.java:395)
	at org.h2.mvstore.db.Store.lambda$new$0(Store.java:125)
	at org.h2.mvstore.MVStore.handleException(MVStore.java:3318)
	at org.h2.mvstore.MVStore.panic(MVStore.java:593)
	at org.h2.mvstore.MVStore.<init>(MVStore.java:469)
	at org.h2.mvstore.MVStore$Builder.open(MVStore.java:4082)
	at org.h2.mvstore.db.Store.<init>(Store.java:136)
	at org.h2.engine.Database.<init>(Database.java:324)
	at org.h2.engine.Engine.openSession(Engine.java:92)
	at org.h2.engine.Engine.openSession(Engine.java:222)
	at org.h2.engine.Engine.createSession(Engine.java:201)
	at org.h2.engine.SessionRemote.connectEmbeddedOrServer(SessionRemote.java:338)
	at org.h2.jdbc.JdbcConnection.<init>(JdbcConnection.java:122)
	at org.h2.Driver.connect(Driver.java:59)
	at java.sql/java.sql.DriverManager.getConnection(DriverManager.java:683)
	at java.sql/java.sql.DriverManager.getConnection(DriverManager.java:230)
	at ru.croc.project.database.access.H2Database.createConnection(H2Database.java:18)
	at ru.croc.project.database.UserDatabaseController.clearDatabase(UserDatabaseController.java:43)
	at ru.croc.project.database.Test.main(Test.java:19)
Caused by: org.h2.jdbc.JdbcSQLNonTransientException: Внутренняя ошибка: "org.h2.mvstore.MVStoreException: The file is locked: D:/programs/Java/learningprogresstracking/TestDatabase/test.mv.db [2.1.214/7]"
General error: "org.h2.mvstore.MVStoreException: The file is locked: D:/programs/Java/learningprogresstracking/TestDatabase/test.mv.db [2.1.214/7]" [50000-214]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:554)
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:477)
	... 20 more
Caused by: org.h2.mvstore.MVStoreException: The file is locked: D:/programs/Java/learningprogresstracking/TestDatabase/test.mv.db [2.1.214/7]
	at org.h2.mvstore.DataUtils.newMVStoreException(DataUtils.java:1004)
	at org.h2.mvstore.FileStore.open(FileStore.java:178)
	at org.h2.mvstore.FileStore.open(FileStore.java:128)
	at org.h2.mvstore.MVStore.<init>(MVStore.java:452)
	... 14 more
2022-12-13 16:07:26 jdbc[3]: exception
org.h2.jdbc.JdbcSQLNonTransientConnectionException: База данных уже закрыта (чтобы отключить автоматическое закрытие базы данных при останове JVM, добавьте ";DB_CLOSE_ON_EXIT=FALSE" в URL)
Database is already closed (to disable automatic closing at VM shutdown, add ";DB_CLOSE_ON_EXIT=FALSE" to the db URL) [90121-214]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:678)
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:477)
	at org.h2.message.DbException.get(DbException.java:223)
	at org.h2.message.DbException.get(DbException.java:199)
	at org.h2.message.DbException.get(DbException.java:188)
	at org.h2.jdbc.JdbcConnection.checkClosed(JdbcConnection.java:1375)
	at org.h2.jdbc.JdbcConnection.prepareStatement(JdbcConnection.java:286)
	at ru.croc.project.database.UserStatisticsDAO.setIntegerStat(UserStatisticsDAO.java:114)
	at ru.croc.project.database.UserStatisticsDAO.setStats(UserStatisticsDAO.java:133)
	at ru.croc.project.database.UserStatisticsDAO.setUserStatistics(UserStatisticsDAO.java:55)
	at ru.croc.project.database.UserDatabaseController.setUserStatistics(UserDatabaseController.java:60)
	at ru.croc.project.test.CommandLineInterface.setUser(CommandLineInterface.java:115)
	at ru.croc.project.test.CommandLineInterface.mainLoop(CommandLineInterface.java:76)
	at ru.croc.project.test.CommandLineInterface.main(CommandLineInterface.java:53)
